src/logging.rs

```diff
@@ -25,7 +25,9 @@ impl log::Log for KernelLogger {
 }
 
 pub fn init() {
-	set_logger(&KernelLogger).expect("Can't initialize logger");
+	unsafe {
+		set_logger_racy(&KernelLogger).expect("Can't initialize logger");
+	}
```

Comment:
This is not sound, as this does not guarantee to uphold the safety requirements.

I think it would be easiest to make `init` unsafe and have the caller uphold the requirements.

Reply:
I fixed it.